addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/handler/GlobalCacheHandler.java

```diff
@@ -621,89 +644,119 @@ protected void connect() {
 
             // If device doesn't have a serial module, just open the command connection
             if (!deviceSupportsSerial1()) {
-                if (deviceConnect(commandConnection)) {
+                if (commandConnect(commandConnection)) {
                     markThingOnline();
                     deviceIsConnected = true;
                     return;
                 }
             } else {
                 // Open the command connection and either 1 or 2 serial connections
-                if (deviceConnect(commandConnection) && deviceConnect(serial1Connection)) {
+                if (commandConnect(commandConnection) && serialConnect(serial1Connection)) {
                     if (deviceSupportsSerial2()) {
-                        if (deviceConnect(serial2Connection)) {
+                        if (serialConnect(serial2Connection)) {
                             markThingOnline();
                             deviceIsConnected = true;
+                            startSerialReaders();
                             return;
                         }
                     } else {
                         markThingOnline();
                         deviceIsConnected = true;
+                        startSerialReaders();
                         return;
                     }
                 }
             }
             disconnect();
         }
 
-        private boolean deviceConnect(DeviceConnection conn) {
+        private boolean commandConnect(DeviceConnection conn) {
```

Comment:
I wanted to clearly differentiate between the connection for handling globalcache commands versus the connection(s) for the serial port(s).  The command port and serial port(s) have distinctly different functionality, so I thought it best to use different terminology.